specification/src/chapter4.adoc

```diff
@@ -2,69 +2,897 @@
+ +
+_Use case examples:_ mobile clients, and automotive.
+
 ==== Isolation model
 
-==== Device access control
+A Global Platform TEE requires the following isolation guarantees:
 
-==== Sealing
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| Root domain MAY access resources assigned to any domain, but SHOULD prevent itself from unintended access to resources assigned to a different domain (privilege escalation).
+
+| CAT_NNN
+| No other domains can access resources assigned to Root domain
+
+| CAT_NNN
+| Resources assigned to TEE domain MUST NOT be accessible to Normal domain
+
+| CAT_NNN
+| Resources assigned to Normal domain MUST be accessible to Normal domain (r/w/x), and to TEE domain (r/w) (default sharing rule)
+
+| CAT_NNN
+| Resources assigned to a single TA, or a guest TEE, MUST not be accessible by a different TA, or guest TEE.
+
+|===
+
+The standard GP TEE model does not support sharing memory between TA in TEE domain. Each TA is expected to be a self-contained unit providing a specific security service, either to Normal domain or to other TA. All communications are implemented through secure channels managed by the TEE OS or SPM. 
+
+Processes in Normal domain can share memory assigned to Normal domain when interacting with a TA in TEE world (default sharing rule). Such shared memory can be cached when context switching between Normal and TEE domains.
+
+RISC-V hardware enforced isolation mechanisms can be used as follows to meet those guarantees:
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN 
+| PMP/ePMP, or MTT, MUST be used to isolate Root domain from other domains.
+
+| CAT_NNN  
+| Supervisor domains MUST be used to enforce isolation between Normal and TEE domains.
+
+|===
+
+See xref:chapter3.adoc#_supervisor_domains[supervisor domains].
+
+For static partition TEE, using PMP/ePMP or PMA with supervisor domains can be sufficient. 
+
+For virtualized TEE, MTT should be used with supervisor domains.
+
+NOTE: MTT can be sufficient for protecting Root domain in the sense that M-mode can enforce that its own resources are never assigned to another domain. PMP/ePMP still add further protections for M-mode, such as the ability to implement temporal isolation boundaries within M-mode (for example, protect early boot code), or to prevent itself from accessing or executing from memory assigned to lower privilege levels (privilege escalation).
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| For a static partition TEE, sPMP or MMU MUST be used to enforce isolation between TA in TEE domain.
+|===
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| For a virtualized TEE, hypervisor extension MUST be supported
```

Comment:
do we consider the case with no Hypervisor, but multiple isolated supervisor domains  ? ree, tee1, tee2 for example